src/driver_humidity.cpp

```diff
@@ -39,4 +70,40 @@ bool HumidityDriver::SendUpdate() {
   return true;
 }
 
+bool HumidityDriver::ProcessConfig(const DriverConfig &config) {
+  HumidityParams humidity_params(config.humidity());
+
+  // Check if calibration is needed
+  if (!humidity_params.do_calibration()) return false;
+
+  // Resetting the calibrated flag
+  calibrated_ = false;
+
+  // Getting temperature data from the humidity sensor
+  matrix_hal::HumidityData data;
+  if (!reader_->Read(&data)) return false;
+  float sensor_temperature = data.temperature;
+
+  // Getting the CPU temperature
+  float cpu_temperature = 0;
+  if (!ReadCpuTemperature(kDefaultCpuTemperatureFileName, &cpu_temperature))
+    return false;
+
+  // Getting the current temperature to use in the calibration
+  const float current_temperature =
+      static_cast<float>(humidity_params.current_temp());
+
+  // Check to avoid devide by zero
+  if (sensor_temperature == cpu_temperature)
```

Comment:
Comparing floats is a tricky thing. It's better if you do:

    #include <cmath>
    ...
    if (std::abs(sensor_temperature - cpu_temperature) < 0.00001f)